elide-core/src/main/java/com/yahoo/elide/core/RequestScope.java

```diff
@@ -102,12 +113,39 @@ protected RequestScope(JsonApiDocument jsonApiDocument, RequestScope outerReques
         this.dictionary = outerRequestScope.dictionary;
         this.mapper = outerRequestScope.mapper;
         this.queryParams = Optional.empty();
+        this.sparseFields = Collections.emptyMap();
         this.objectEntityCache = outerRequestScope.objectEntityCache;
         this.securityMode = outerRequestScope.securityMode;
         this.deferredChecks = outerRequestScope.deferredChecks;
     }
 
     /**
+     * Parses queryParams and produces sparseFields map
+     * @param queryParams The request query parameters
+     * @return Parsed sparseFields map
+     */
+    private static Map<String, Set<String>> parseSparseFields(MultivaluedMap<String, String> queryParams) {
+        Map<String, Set<String>> result = new HashMap<>();
+
+        for (String key : queryParams.keySet()) {
```

Comment:
perhaps use queryParams.entrySet() here so you have efficient access to values later.